tools/setup/emoji/emoji_setup_utils.py

```diff
@@ -223,3 +223,19 @@ def emoji_names_for_picker(emoji_map):
         codepoint_to_names[codepoint] = names
 
     return sorted(list(chain.from_iterable(codepoint_to_names.values())))
+
+def generate_emoji_catalog(emoji_data):
+    # type: (List[Dict[Text, Any]]) -> Dict[str, List[str]]
+    sort_order = {} # type: Dict[str, int]
+    emoji_catalog = {} # type: Dict[str, List[str]]
```

Comment:
There is some problem in stringifying a `defaultdict()`, so I am leaving it as it is.